src/plugins/analysis/known_vulnerabilities/test/test_known_vulnerabilities.py

```diff
@@ -1,94 +1,87 @@
-# pylint: disable=protected-access
-
 import json
 import os
 
+import pytest
 from common_helper_files import get_dir_of_file
 
 from objects.file import FileObject
 from plugins.analysis.known_vulnerabilities.code.known_vulnerabilities import AnalysisPlugin
-from test.unit.analysis.analysis_plugin_test_class import AnalysisPluginTest  # pylint: disable=wrong-import-order
 
 TEST_DATA_DIR = os.path.join(get_dir_of_file(__file__), 'data')
 
 
-class TestAnalysisPluginsKnownVulnerabilities(AnalysisPluginTest):
-
-    PLUGIN_NAME = 'known_vulnerabilities'
-    PLUGIN_CLASS = AnalysisPlugin
-
-    def setUp(self):
-        super().setUp()
-        with open(os.path.join(TEST_DATA_DIR, 'sc.json')) as json_file:
-            self._software_components_result = json.load(json_file)
+@pytest.mark.AnalysisPluginClass.with_args(AnalysisPlugin)
+class TestAnalysisPluginsKnownVulnerabilities:
+    with open(os.path.join(TEST_DATA_DIR, 'sc.json'), 'r') as json_file:
+        _software_components_result = json.load(json_file)
```

Comment:
While it's technically not illegal, I don't think this code should just sit at the top of this class. But we could convert it to a "pure" class constant assignment by replacing os.path with pathlib:
```suggestion
    _software_components_result = json.loads((Path(TEST_DATA_DIR) / 'sc.json').read_text())
```

Reply:
Maybe in a future PR we can provide a fixture named sth. along the lines of `test_data_file` that can be parameterized to give just the file needed.

For now I changed the code to your suggestion.